interview-questions.md

```diff
@@ -4,49 +4,85 @@ Answer the following questions. First, without external resources. Challenge you
+
+  A virtual DOM object is a representation of a DOM object, like a lightweight copy. A virtual DOM object has the same properties as a real DOM object, but it lacks the real thing’s power to directly change what’s on the screen. In summary, here’s what happens when you try to update the DOM in React:
+
+  1. The entire virtual DOM gets updated.
+  2. The virtual DOM gets compared to what it looked like before you updated it. React figures out which objects have changed.
+  3. The changed objects, and the changed objects only, get updated on the real DOM.
+  4. Changes on the real DOM cause the screen to change.
+
+  (https://www.codecademy.com/articles/react-virtual-dom)
 
-  Researched answer:
 
 
 
 3. When creating a basic (stateless) class component in React, what are the necessary elements needed to display "Hello World" in the browser?
 
   Your answer:
+   A class, that extends a react component, a render,  a return, and content "hello world",  export default
 
   Researched answer:
+    // We're using React and Component from the react library
+    import React, { Component } from 'react'
 
+    // We create a new component that extends the Component class, which React provides
 
+    class App extends Component{
+    // When the component is first put on the page, and every time something changes in our component, the render function is called
+    // This function builds the JSX that will be put on the page by React
+      render(){
+        // Most of the React apps use something called JSX to preprocess HTML (XML actually) back into plain old JavaScript before the file is sent to the browser
 
-4. What is JSX?
+        return (
+          <h1>A Basic Class Component</h1>
+        )
+      }
+    }
+    // Every component must be exported
+    export default App
 
-  Your answer:
+    (https://github.com/LEARNAcademy/Syllabus/blob/master/react/create-react-app.md)
 
-  Researched answer:
 
 
+4. What is JSX?
+
+  Your answer: JavaScript XML markup language is used with React to display things to a web page
+
+  Researched answer:  JSX is JavaScript flavored HTML that behaves almost like HTML with a couple small exceptions (https://github.com/LEARNAcademy/Syllabus/blob/master/react/nested-components.md) JSX is used in React apps to preprocess HTML (XML actually) back into plain old JavaScript before the file is sent to the browser (https://github.com/LEARNAcademy/Syllabus/blob/master/react/create-react-app.mdß)
+ 
+
 
 5. What is state in React?
 
-  Your answer:
+  Your answer: A state is the current data in a react component that changes dynamically.
+
+  Researched answer: State is an object that stores a component's dynamic data and determines how the component renders and behaves. (https://github.com/LEARNAcademy/Syllabus/blob/master/react/state.md)
+
 
-  Researched answer:
 
 
 
 6. STRETCH: What is hoisting in JavaScript?
 
-  Your answer:
+  Your answer: No idea!
```

Comment:
Haha, totally acceptable answer! I love the honesty. 

You'll hear this from Bryan in PD week and in interview practice, but I'd like you to get into the habit early of saying something along the lines of "Not sure, but heres how I would find out and what I'd research.." to show the interviewer that just because you may not know now, you're resourceful in finding the answer.